Launch new postgrest orchestration

johnclary opened the pull request:
This PR serves as a place to review this new system for managing our slew of postgREST instances. It is currently deployed and active.

I'm looking for feedback on:
- The documentation in this repo
- The documentation in our password manager, which can be found by searching "atd-postgrest".

There is currently no way to fully test this. We do not currently have test instances for any of the DBs. I'm very open to working on that. Each of the various projects that rely on these services will have their own unique constraints around test environment. 

As a minimum test, it would be good if you verify that you can connect to the EC2 bastion host:
1. SSH into the host (see 1Pass)
2. `$ sudo docker ps` to observe the containers.

frankhereford reviewed (approved):
@johnclary You did an awesome job with this configuration and migration. Genuinely seamless work. Thank you! 🙏🏻

I think your documentation is really sharp - it's accessible and detailed. 

My thoughts below are just a notepad of ideas. They shouldn't stop this from being merged and this service to be acknowledged as permanent in any way. 

Here are a few things that have been helpful to me in the past that you might consider:
* If you have a working installation running on an EC2 instance, you might make an AMI of that which will always be available for us to spin up a replacement instance if that one were to kernel panic or otherwise go into a running-but-unresponsive state. When we make meaningful updates to the EC2 instance, such as deploying system updates or new code from the repo, refreshing that AMI should be part of the task. This also has the benefit of freezing any secrets you may have on the filesystem but not checked into the repo (of course) in place and ready to re-spin up the docker-compose stack.
* Check to see if you can reboot the machine. Does it come up and start its services? Docker-compose sometimes (at least in my hair-brained configurations) will not offer DNS resolution of sibling containers until they hit an up and running state. This angers HAProxy, which I am 50/50 being a bug or a feature, but none-the-less, causes it to give up and exit().
* Update process - is there any? I think this repo is serving up a middleware type solution, but would we need to do anything to the postgREST processes if there were schema changes on the RDS servers.

johnclary commented:
Thanks for your helpful feedback @frankhereford!
> make an AMI of that which will always be available 
This is a great idea! I will work on it.

> Check to see if you can reboot the machine. Does it come up and start its services
Excellent question. I've now configured docker to start on boot. And confirmed that services came back up after reboot. I've also added a keep-alive script, which is not foolproof but may help mitigate any random outages. 
See the [deployment](https://github.com/cityofaustin/atd-postgrest/tree/jc-launch#deployment) section of the readme.

> Update process - is there any? 
I've updated the readme about this. 

Happy to refine any of this further—let me know what you think!

mateoclarke reviewed (approved):
🚀
